src/gpu/SDL_gpu.c

```diff
@@ -330,12 +330,18 @@ static SDL_GpuDriver SDL_GpuSelectBackend(
     /* Environment/Properties override... */
     if (gpudriver != NULL) {
         for (i = 0; backends[i]; i += 1) {
-            if (SDL_strcasecmp(gpudriver, backends[i]->Name) == 0 && (backends[i]->shaderFormats & formatFlags) && backends[i]->PrepareDriver(_this)) {
-                return backends[i]->backendflag;
+            if (SDL_strcasecmp(gpudriver, backends[i]->Name) == 0) {
+                if (!(backends[i]->shaderFormats & formatFlags)) {
+                    SDL_LogError(SDL_LOG_CATEGORY_GPU, "Unsupported shader format for backend %s!", gpudriver);
```

Comment:
This message could be clearer - like "Required shader format for backend %s not provided!"

Reply:
Updated